### PR TITLE
tests(feat): add tests to advisory-list

### PR DESCRIPTION
**Pull request by carlosthe19916**
## Summary by Sourcery

Add shared E2E filter testing helpers and extend advisory and SBOM list filter coverage, including state persistence and edge cases.

Enhancements:
- Update toolbar chip removal and label assertions to support precise label matching and detection of an empty filter state.
- Adjust SBOM list license filter to use a typeahead control and set the autocomplete select role for accessibility and testing.

Tests:
- Add reusable Playwright helper utilities for filter behavior, empty state, clear-all, chip removal, and URL/navigation persistence across list pages.
- Expand SBOM list filter tests with partial matches, edge cases, and filter state persistence scenarios.
- Add comprehensive advisory list filter tests covering partial search, edge conditions, clear-all, chip/group removal, and URL/navigation persistence.

### CODE REVIEW

File: e2e/tests/ui/pages/advisory-list/filter.spec.ts [hunk 1]
```diff
@@ -30,4 +31,475 @@ test.describe("Filter validations", { tag: "@tier1" }, () => {
...
+      Revision: { from: "01/01/2024", to: "12/31/2025" },
+      Label: ["type=cve"],
+    });
+    await expect(table).toHaveNumberOfRows({ lessThan: 5 });
+
+    // Clear all filters
+    await toolbar.clearAllFilters();
+
+    // Should show more results (not filtered)
+    await expect(table).toHaveNumberOfRows({ equal: 10 });
```
[the remark this answers is not in the document]
Comment (carlosthe19916) [reply e2e/tests/ui/pages/advisory-list/filter.spec.ts]: toHaveNumberOfRows does what `getRowCount` is supposed to do

**Review by stanislavsemeniuk — APPROVED** [review]
